src/components/Conference.tsx

```diff
@@ -153,16 +153,20 @@ export class Conference extends React.Component<IConferenceProps, IConferenceSta
             return;
         };
 
-        const connection = this.props.connect();
-        if (!connection) {
-            this.onError(createConferenceErrorConnect())
-            return
-        }
-        this.connection = connection
-        this.joinRoom(this.props.room);
-        this.getUserMedia().then(() => {
-            this.connection.subscribe(this.handleIncomingMessage)
-        });
+        this.connection = this.props.connect();
+
+        // NOTE(yunsi): Create websocket connection, if succeed then join room, if failed then fire error.
+        this.connection.connect()
+            .then(() => {
+                this.joinRoom(this.props.room);
```

Comment:
The logic here will be an issue.

joinRoom and getUserMedia should be Promise-Based API and put in the pipeline.

Reply:
Add a TODO here

Reply:
Added TODO